### ISSUE TITLE
Incorrect color conversion to grayscale

### ISSUE THREAD

**Issue by hilbigan** [SerenityOS/serenity#9758]
Colors are converted to grayscale by averaging the three channels:

https://github.com/SerenityOS/serenity/blob/beb8c48f26c882a77d7ddf323463b6f0f07e08f0/Userland/Libraries/LibGfx/Color.h#L226

Because the human eye is more sensitive to green, images are usually
converted to grayscale using [the luminosity formula](https://en.wikipedia.org/wiki/Grayscale#Luma_coding_in_video_systems) instead.

This produces a better result, [see here](https://www.johndcook.com/blog/2009/08/24/algorithms-convert-color-grayscale/) for a comparison.

The code would need to be changed to something like this:

```c++
int gray = (int) (0.299 * (float) red() + 0.587 * (float) green() + 0.114 * (float) blue()); 
```

Edit: This might be a matter of preference, but the luminosity eqn. should be preferred for image
processing, e.g. in the PixelPaint grayscale filter.

**Comment by mustafaquraish** [comment]
If you want to be technical about it, the formula you've provided is also not necessarily correct. You need to linearize the values before applying the coefficients (Read more: https://stackoverflow.com/a/56678483/13782556)

In any case, that's probably a bit too far down the rabbit hole, though I agree we should probably use a better way to convert RGB to Grayscale.

**Comment by mustafaquraish** [comment]
Quick turnaround, fix has been merged :^)

**Comment by alimpfard** [comment]
Fixed by #9763.
